src/main.cpp

```diff
@@ -2945,8 +2945,7 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
                         LogPrint("parallel", "Sending Quit() to scriptcheckqueue\n");
                         (*mi).second.pScriptQueue->Quit();
                     }
-                    //(*mi).second.tRef->interrupt(); // interrupt the thread
-                    (*mi).second.fQuit = true; // interrupt the thread
+                    (*mi).second.fQuit = true; // quit the thread
```

Comment:
Updated and changed to Atomic.

Two mechanisms are used to exit the PV thread...one is to interrupt the scriptcheck thread pool (used for checking sigs) the other is to send a quit to the PV thread itself.